### DISCUSSION TITLE
Upgrading without NC_AUTH_JWT_SECRET set

### DISCUSSION THREAD

**Discussion by just-grizzle** [nocodb/nocodb#7259] [General]
### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Steps to reproduce ?

Hi, I deployed Noco with Docker. But I did not set the  `NC_AUTH_JWT_SECRET` env variable.

### Desired Behavior

I'd like to upgrade NocoDB to the latest version. Reading that people got messed up installations when their `NC_AUTH_JWT_SECRET` changed, I wondered what happened if a random secret was generated.

Can I upgrade and are there any considerations to be taken?

### Project Details

Node: **v12.22.12**
Arch: **arm64**
Platform: **linux**
Docker: **true**
Database: **pg**
ProjectOnRootDB: **true**
RootDB: **pg**
PackageVersion: **0.92.4**

**Comment by pranavxc** [comment]
You have to worry only if you set `NC_AUTH_JWT_SECRET` environment variable, otherwise we populate a random token and persist in meta database, on upgrade it will choose from meta db. If you set `NC_AUTH_JWT_SECRET` earlier and removed while upgrading then it's going to break, existing project config(connection config) are encrypted with the same token and no longer able to decrypt since token changed, and any new project will use the new token in this scenario.

**Comment by just-grizzle** [comment]
Upgrade went through perfectly. Thx a lot.

Out of curiosity: Is it possible to read out the random JWT secret and copy it to my Docker compose file for future use?

**Reply by pranavxc** [reply]
Yes you can get it from `nc_store` table.
